### PR TITLE
[CUMULUS-417] Update ESLint configuration

### PR DESCRIPTION
**Pull request by bt6453**
**Summary:** Summary of changes

Addresses [CUMULUS-417: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-417)

## Changes

* Add ESLint config to closer match the core repo
* Add Airbnb suggestions
* Add eslint-comments suggestions
* Add import suggestions
* Add lodash suggestions

## PR Checklist

- [x] Update CHANGELOG
- [X] Unit tests
- [X] Adhoc testing
- [X] Integration tests

### CODE REVIEW

File: app/src/js/middleware/validate.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { CALL_API } from '../actions/types';
 
-export const isValidApiRequestAction = (action) => {
-  return typeof action === 'object' && Object.prototype.hasOwnProperty.call(action, CALL_API);
-};
+export const isValidApiRequestAction = (action) => typeof action === 'object' && Object.prototype.hasOwnProperty.call(action, CALL_API);
```
Comment (flamingbear) [inline app/src/js/middleware/validate.js:3]: ugh.  this is just a note to see if we can modify the arrow rule.

File: app/src/js/components/Executions/execution-status-graph.js [hunk 2]
```diff
@@ -110,4 +108,4 @@ ExecutionStatusGraph.propTypes = {
   executionStatus: PropTypes.object
 };
 
-export default connect(state => ({}))(ExecutionStatusGraph);
+export default connect(() => ({}))(ExecutionStatusGraph);
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply app/src/js/components/Executions/execution-status-graph.js]: I unconnected it.

File: app/src/js/components/Executions/overview.js [hunk 3]
```diff
@@ -60,7 +57,7 @@ class ExecutionOverview extends React.Component {
     const { dropdowns } = collections;
     const { list } = executions;
     const { count, queriedAt } = list.meta;
-    if (list.infix && list.infix.value) {
+    if (list.asyncOperationId && list.asyncOperationId.value) {
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply app/src/js/components/Executions/overview.js:60]: @jennyhliu or @dopeters Can you look to see why this change does or does not make sense, in terms of lint?

[the remark this answers is not in the document]
Comment (dopeters) [reply app/src/js/components/Executions/overview.js:60]: So, as far as I'm concerned, this line doesn't make sense regardless. I think this is supposed to be filtering the executions based on the search term (i.e. infix) that is entered. In the executions reducer, we have 

```
[SEARCH_EXECUTIONS]: (state, action) => {
  state.list.infix = action.infix;
},
[CLEAR_EXECUTIONS_SEARCH]: (state) => {
  state.list.infix = null;
},
```
So we should be checking for `list.infix` here. However, `list.infix.value` will never be true since we're just setting a string (`action.infix` is just a string of the search query, not an object with a value param)

Basically, that line will never be true either way, so 🤷. I _think_ based on previous discussions, we may be able to just get rid of that entirely, including the `searchOperationId` function, but would be interested in @jennyhliu's input

[the remark this answers is not in the document]
Comment (jennyhliu) [reply app/src/js/components/Executions/overview.js:60]: Yeah.  The change doesn't make sense.  We could get rid of that part.  Maybe a separate PR and also add cypress test.
Search with asyncOperationId works as the following, and infix is not used. 
executions?limit=50&page=1&sort_key%5B%5D=-createdAt&asyncOperationId=myid

[the remark this answers is not in the document]
Comment (jennyhliu) [reply app/src/js/components/Executions/overview.js:60]: I am not sure how partial search with asyncOperationId would work.  Maybe a separate ticket to look into more and cleanup the code.

[the remark this answers is not in the document]
Comment (flamingbear) [reply app/src/js/components/Executions/overview.js:60]: @jennyhliu would you please write that ticket since you understand better than me?  Thank you.

[the remark this answers is not in the document]
Comment (flamingbear) [reply app/src/js/components/Executions/overview.js:60]: I'll revert this for the PR.

[the remark this answers is not in the document]
Comment (jennyhliu) [reply app/src/js/components/Executions/overview.js]: https://bugs.earthdata.nasa.gov/browse/CUMULUS-2134

[the remark this answers is not in the document]
Comment (flamingbear) [reply app/src/js/components/Executions/overview.js]: 🙌 thanks

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+test/*
```
[the remark this answers is not in the document]
Comment (flamingbear) [reply .eslintignore:1]: Made changes to allow expanding to include test and cypress.

File: app/src/js/actions/index.js [hunk 5]
```diff
@@ -1,26 +1,25 @@
-'use strict';
-
+/* eslint-disable import/no-cycle */
```
[the remark this answers is not in the document]
Comment (bt6453) [reply app/src/js/actions/index.js:1]: Yeah I was going to take a crack at the cyclic imports but ran out of time. Thanks for making a ticket.

File: app/src/js/config/index.js [hunk 6]
```diff
@@ -35,6 +34,7 @@ const baseConfig = {
   updateDelay: 1000
 };
 
+// eslint-disable-next-line prefer-object-spread
```
[the remark this answers is not in the document]
Comment (bt6453) [reply app/src/js/config/index.js:37]: You saw this and I'm six days too late, but: it breaks things.
